server/chat/backend/agent/orchestrator/select_skills.py

```
@@ -0,0 +1,256 @@
+        connected_ids = set(registry.get_connected_skill_ids(user_id) or [])
+        for skill_id, meta in registry._skills.items():
+            for tool_name in (meta.tools or []):
+                skill_owned.add(tool_name)
+                if skill_id in connected_ids:
+                    connected.add(tool_name)
+    except Exception:
+        logger.exception("select_skills: failed to resolve connected-tool filter for user")
+    return skill_owned, connected
+
+
+def select_tools_for_role(user_id: str, role: Any, all_tools: list) -> list:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `role`

(ANN401)